### ISSUE TITLE
proposal: interval reporting for network access loggers

### ISSUE THREAD

**Issue by kyessenov** [envoyproxy/envoy#10936]
Network access loggers can be attached to listeners and tcp_proxies. They report once the connection is closed. This is a problem for long-running, stalled, and slow connections, since there is no telemetry until it's over, which complicates debugging of connection handling issues.

This proposal is to add an option, e.g. `interval_duration: 10s`, to invoke the access loggers `log()` function repeatedly (and immediately on the connection establishment). This means some data might not be populated in StreamInfo, so the access log needs to distinguish between missing and not-yet-populated states.

The alternative is to push down the functionality into the access log extension itself. For that to be possible:
1. We would need ability to set up timers from FactoryContext.
2. The latter point stands: we should be able to distinguish between missing and not yet defined.

**Comment by kyessenov** [comment]
cc @jplevyak for thoughts how that would interact with Wasm shim.

**Comment by mattklein123** [comment]
+1 I like this idea. The other thing I would love you to implement as part of this change is an option to log both at the beginning and end of the connection. So a variant of what you are proposing. Can you take a look at that also?

**Comment by kyessenov** [comment]
Yeah, that shouldn't be too hard to report on connection open in the connection handler.
